commit 9fed8d7c46f37151037334ef5e8b30b945baaceb
Author: Finn Callies <fcallies@linux.ibm.com>
Date:   Wed Sep 11 09:21:09 2024 +0200

    s390/crypto: Display Query and Query Authentication Information in sysfs
    
    Displays the query (fc=0) and query authentication information (fc=127)
    as binary in sysfs per CPACF instruction. Files are located in
    /sys/devices/system/cpu/cpacf/. These information can be fetched via
    asm already except for PCKMO because this instruction is privileged. To
    offer a unified interface all CPACF instructions will have this
    information displayed in sysfs in files <instruction>_query_raw and
    <instruction>_query_auth_info_raw.
    
    A new tool introduced into s390-tools called cpacfinfo will use this
    information to convert and display in human readable form.
    
    Suggested-by: Harald Freudenberger <freude@linux.ibm.com>
    Reviewed-by: Harald Freudenberger <freude@linux.ibm.com>
    Acked-by: Heiko Carstens <hca@linux.ibm.com>
    Signed-off-by: Finn Callies <fcallies@linux.ibm.com>
    Signed-off-by: Vasily Gorbik <gor@linux.ibm.com>

diff --git a/arch/s390/kernel/Makefile b/arch/s390/kernel/Makefile
index 5ceb08b338d3..48caae8c7e10 100644
--- a/arch/s390/kernel/Makefile
+++ b/arch/s390/kernel/Makefile
@@ -50,6 +50,7 @@ extra-y				+= vmlinux.lds
 obj-$(CONFIG_SYSFS)		+= nospec-sysfs.o
 CFLAGS_REMOVE_nospec-branch.o	+= $(CC_FLAGS_EXPOLINE)
 
+obj-$(CONFIG_SYSFS)		+= cpacf.o
 obj-$(CONFIG_MODULES)		+= module.o
 obj-$(CONFIG_SCHED_TOPOLOGY)	+= topology.o hiperdispatch.o
 obj-$(CONFIG_NUMA)		+= numa.o